hudi-common/src/main/java/org/apache/hudi/common/table/view/AbstractTableFileSystemView.java

```diff
@@ -915,7 +963,15 @@ public final Option<FileSlice> getLatestFileSlice(String partitionStr, String fi
         if (!fs.isPresent()) {
           return Option.empty();
         }
-        return Option.ofNullable(filterUncommittedFiles(fs.get(), true).map(this::addBootstrapBaseFileIfPresent).findFirst().orElse(null));
+        Stream<FileSlice> fileSlices = tableVersion8AndAbove()
+            ? this.filterUncommittedFiles(fs.get(), true)
+            : this.filterBaseFileAfterPendingCompaction(fs.get(), true);
```

Comment:
Do we need to also fix the merged log record scanner to bring back the pending instant check for the log blocks because of these unfiltered log files?